### PR TITLE
#307615 Retrieve a presigned URL for a file

### PR DESCRIPTION
**Pull request by alexluckett**
Decouples this service from the wider forms ecosystem by removing the need to store the formId and then do a further lookup in forms-manager. Instead, we store a retrievalKey which is provided by the process initiator (forms-runner) and then the user (when retrieving a file from forms-designer).

This stores an encrypted retrievalKey per file using the OSWAP-recommended argon2id algorithm during ingestion, which is then used to retrieve the file later on. This prevents a user changing the output email address in forms-designer and gaining access to previously uploaded files - they must know the value at the time of upload.

### CODE REVIEW

File: src/api/files/service.js [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
     await repository.create({
-      formId,
-      ...fileContainer
+      ...fileContainer,
+      retrievalKey: hashed
     })
   } catch (err) {
     if (err instanceof MongoServerError && err.errorResponse.code === 11000) {
-      const error = `File ID '${fileContainer.fileId}' for form ID '${formId}' has already been ingested`
+      const error = `File ID '${fileContainer.fileId}' has has already been ingested`
```
Comment (KiranVarma02) [inline src/api/files/service.js:32]: 2 * has :)

File: src/routes/files.test.js [hunk 2]
```diff
@@ -72,6 +78,28 @@ describe('Forms route', () => {
...
+          fileId: '1234',
+          retrievalKey: 'test'
+        }
+      })
+
+      expect(response.statusCode).toEqual(StatusCodes.OK)
+      expect(response.headers['content-type']).toContain('application/json')
+      expect(response.result).toMatchObject({
+        url: expect.any(String)
+      })
```
Comment (KiranVarma02) [inline src/routes/files.test.js:101]:
```suggestion
        url: 'https://s3.dummy.com/file.txt'
      })
```

File: src/api/files/service.js (hunk 1 above)
Comment (alexluckett) [reply src/api/files/service.js]: Fixed in [9f4f60e](https://github.com/DEFRA/forms-submission-api/pull/8/commits/9f4f60e31dce89fac9821bddf5d1f17d07a07fb2), thanks.

File: src/routes/files.test.js (hunk 2 above)
Comment (alexluckett) [reply src/routes/files.test.js:101]: Fixed in [c7ee98f](https://github.com/DEFRA/forms-submission-api/pull/8/commits/c7ee98f627c6542a5999c0fbe68362296483ac48), thanks.

**Review by KiranVarma02 — APPROVED** [review]

**Pull request merged**